src/main/java/Command.java

```diff
@@ -0,0 +1,170 @@
+import java.io.IOException;
+
+class Command {
+
+    /**
+     * constructor.
+     * @param tasks is the list of tasks.
+     * @param ui is the UI class.
+     * @param storage is the Storage class.
+     * @throws DukeException when input is different from required.
+     * @throws IOException when input is different from required.
+     */
+    void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException, IOException{
+
+    }
+
+    /**
+     * whether is it an exit command
+     * @return true when it is an exit command, false otherwise.
+     */
+    boolean isExit() {
+        return (this instanceof ExitCommand);
+    }
+}
+
+class AddCommand extends Command {
+    /**
+     * command to add a task.
+     */
+    private Task t;
+
+    /**
+     * constructor.
+     * @param t is the task.
+     */
+    AddCommand(Task t) {
+        this.t = t;
+    }
+
+    /**
+     * add task and print out information.
+     * @param tasks is the list of tasks.
+     * @param ui is the UI class.
+     * @param storage is the Storage class.
+     * @throws IOException when input is different from required.
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws IOException {
+        tasks.addTask(t);
+        storage.save();
+        System.out.println("     Got it. I've added this task:");
```

Comment:
Have you considered passing such interactions to the responsibility of the `Ui` class instead? Might be better to maintain that all printing is done by the same class.